internal/auth0/organization/resource_member.go

```diff
@@ -67,101 +51,11 @@ func createOrganizationMember(ctx context.Context, d *schema.ResourceData, m int
 
 	d.SetId(organizationID + ":" + userID)
 
-	if err := assignMemberRoles(ctx, d, m); err != nil {
-		if err, ok := err.(management.Error); ok && err.Status() == http.StatusNotFound {
-			d.SetId("")
-			return nil
-		}
-
-		return diag.FromErr(err)
-	}
-
 	return readOrganizationMember(ctx, d, m)
 }
 
-func assignMemberRoles(ctx context.Context, d *schema.ResourceData, meta interface{}) error {
-	if !d.HasChange("roles") {
-		return nil
-	}
-
-	userID := d.Get("user_id").(string)
-	organizationID := d.Get("organization_id").(string)
-
-	toAdd, toRemove := value.Difference(d, "roles")
-
-	if err := removeMemberRoles(ctx, meta, organizationID, userID, toRemove); err != nil {
-		return err
-	}
-
-	return addMemberRoles(ctx, meta, organizationID, userID, toAdd)
-}
-
-func removeMemberRoles(ctx context.Context, meta interface{}, organizationID string, userID string, roles []interface{}) error {
-	if len(roles) == 0 {
-		return nil
-	}
-
-	var rolesToRemove []string
-	for _, role := range roles {
-		rolesToRemove = append(rolesToRemove, role.(string))
-	}
-
-	api := meta.(*config.Config).GetAPI()
-
-	return api.Organization.DeleteMemberRoles(ctx, organizationID, userID, rolesToRemove)
-}
-
-func addMemberRoles(ctx context.Context, meta interface{}, organizationID string, userID string, roles []interface{}) error {
-	if len(roles) == 0 {
-		return nil
-	}
-
-	var rolesToAssign []string
-	for _, role := range roles {
-		rolesToAssign = append(rolesToAssign, role.(string))
-	}
-
-	api := meta.(*config.Config).GetAPI()
-
-	return api.Organization.AssignMemberRoles(ctx, organizationID, userID, rolesToAssign)
-}
-
-func readOrganizationMember(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-	api := m.(*config.Config).GetAPI()
-
-	organizationID := d.Get("organization_id").(string)
-	userID := d.Get("user_id").(string)
-
-	roles, err := api.Organization.MemberRoles(ctx, organizationID, userID)
-	if err != nil {
-		if err, ok := err.(management.Error); ok && err.Status() == http.StatusNotFound {
-			d.SetId("")
-			return nil
-		}
-		return diag.FromErr(err)
-	}
-
-	var rolesToSet []string
-	for _, role := range roles.Roles {
-		rolesToSet = append(rolesToSet, role.GetID())
-	}
-
-	err = d.Set("roles", rolesToSet)
-
-	return diag.FromErr(err)
-}
-
-func updateOrganizationMember(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-	if err := assignMemberRoles(ctx, d, m); err != nil {
-		if err, ok := err.(management.Error); ok && err.Status() == http.StatusNotFound {
-			d.SetId("")
-			return nil
-		}
-
-		return diag.FromErr(err)
-	}
-
-	return readOrganizationMember(ctx, d, m)
+func readOrganizationMember(_ context.Context, _ *schema.ResourceData, _ interface{}) diag.Diagnostics {
+	return nil
 }
```

Comment:
Just curious, why can't this function be removed?

Reply:
It's not optional, every resource needs to have a read defined. Otherwise we get an error saying: ``` Read must be implemented ```.